engine/robot_logic/traversal.py

```diff
@@ -308,3 +308,36 @@ def turn_to_target_heading(
         abs_heading_error = abs(target_heading - float(predicted_state[2]))
     # re-enable after finishing turning
     robot_state.enable_obstacle_avoidance = True
+
+# Simpler version of the funcrtion above without using PID
+def simpler_turn_to_target_heading(
+    robot_state, target_heading, allowed_heading_error, database
+):
+    """
+    Turns robot in-place to target heading + or - allowed_heading_error
+        target_heading: the heading in radians the robot should approach at the end of in-place rotation.
+        allowed_heading_error: the maximum error in radians a robot can have to target heading while turning in
+            place.
+    """
+    # we dont want the robot to avoid obstacle here
+    robot_state.enable_obstacle_avoidance = False
+
+    while abs(target_heading - float(robot_state.state[2])) > allowed_heading_error:
+        theta_error = target_heading - robot_state.state[2]
```

Comment:
TODO: we can probably optimize this such that the robot turns the shorter angle (imagine target_heading is 359 degrees and robot_state.state[2] is 1, then it is more optimal to turn from 359 to 360 to 1 than 359 to 358 ... to 1